File: backend/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Order extends Model
{
    use HasFactory;

    protected $fillable = [
        'customer_name',
        'customer_phone',
        'wilaya',
        'commune',
        'address',
        'product_name',
        'quantity',
        'unit_price',
        'total_price',
        'status',
        'assigned_to',
        'notes',
        'call_attempts',
        'ayor_order_id',
        'category_id',
        'tracking_number',
        'shipping_company',
        'shipping_status',
        'delivery_type',
        'stopdesk_id',
        'postponed_date',
    ];

    protected $casts = [
        'unit_price' => 'decimal:2',
        'total_price' => 'decimal:2',
        'quantity' => 'integer',
        'call_attempts' => 'integer',
        'category_id' => 'integer',
        'postponed_date' => 'date',
    ];

    // Relationships
    public function agent()
    {
        return $this->belongsTo(User::class, 'assigned_to');
    }

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function callLogs()
    {
        return $this->hasMany(CallLog::class);
    }
}

File: mobile/lib/features/orders/models/order.dart
class Order {
  final int id;
  final String customerName;
  final String customerPhone;
  final String wilaya;
  final String? commune;
  final String? address;
  final String productName;
  final int quantity;
  final double totalPrice;
  final String status;
  final String? trackingNumber;
  final String? notes;
  final DateTime createdAt;

  Order({
    required this.id,
    required this.customerName,
    required this.customerPhone,
    required this.wilaya,
    this.commune,
    this.address,
    required this.productName,
    required this.quantity,
    required this.totalPrice,
    required this.status,
    this.trackingNumber,
    this.notes,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'],
      customerName: json['customer_name'] ?? 'Inconnu',
      customerPhone: json['customer_phone'] ?? '',
      wilaya: json['wilaya'] ?? '',
      commune: json['commune'],
      address: json['address'],
      productName: json['product_name'] ?? '',
      quantity: json['quantity'] ?? 1,
      totalPrice: double.tryParse(json['total_price']?.toString() ?? '0') ?? 0.0,
      status: json['status'] ?? 'new',
      trackingNumber: json['tracking_number'],
      notes: json['notes'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}
